data_loader.py

```diff
@@ -28,45 +31,103 @@ def disconnect(self):
         # Nothing to do
         pass
 
+    def _flatten_dict(self, data: dict, keystring: str = ""):
```

Comment:
It's not flattening `dict` but it transforms `dict` to `tuple` :)
So could be better to either returns flatten `dict` or change name of the function to more appropriate :)